src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -617,7 +617,7 @@ public void savePredicates() throws IOException {
     for (String session : sessions.keySet()) {
 
       // TODO: better parsing of this.
-      String[] parts = session.split("-");
+      String[] parts = session.split("_");

```

Comment:
Rather than fixing it everywhere else, why not just address the TODO here...

Reply:
The idea is that the session should have a proper encoding / decoding so it doesn't have to do a blind splitting on a character.  The splitting character should be escaped and obeyed.

Reply:
ok will find Something cleaner